### PR TITLE
Move text examples out of pylab_examples

### PR DESCRIPTION
**Pull request by dstansby**
Some more pylab_examples moves, this time what I think are all the text related examples.

### CODE REVIEW

**Comment by QuLogic** [comment]
Sorry, some of this conflicted with #8199.

File: examples/text_labels_and_annotations/tex_demo.py [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
 parameter text.usetex is set.  This works currently on the agg and ps
 backends, and requires that you have tex and the other dependencies
 described at http://matplotlib.org/users/usetex.html
 properly installed on your system.  The first time you run a script
 you will see a lot of output from tex and associated tools.  The next
-time, the run may be silent, as a lot of the information is cached.
-
-Notice how the the label for the y axis is provided using unicode!
+time, the run may be silent, as a lot of the information is cached in
+~/.tex.cache
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/tex_demo.py]: This should not be here; I think you may have resolved conflicts in the opposite direction.

Comment (dstansby) [inline examples/text_labels_and_annotations/tex_demo.py]: I've just put back in the master copy of tex_demo.py, which I think is correct.

File: examples/text_labels_and_annotations/mathtext_examples.py [hunk 2]
```diff
@@ -59,9 +59,9 @@
 
 def doall():
     # Colors used in mpl online documentation.
-    mpl_blue_rvb = (191./255., 209./256., 212./255.)
-    mpl_orange_rvb = (202/255., 121/256., 0./255.)
-    mpl_grey_rvb = (51./255., 51./255., 51./255.)
+    mpl_blue_rvb = (191 / 255, 209 / 256, 212 / 255)
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/mathtext_examples.py]: Needs `from __future__ import division` for this to work on Python 2.

File: examples/text_labels_and_annotations/mathtext_examples.py [hunk 3]
```diff
@@ -79,25 +79,25 @@ def doall():
...
+                 xy=(0.5, 1. - 0.59 * line_axesfrac),
                  xycoords='data', color=mpl_orange_rvb, ha='center',
                  fontsize=20)
 
     # Plotting features demonstration formulae
     for i_line in range(1, n_lines):
-        baseline = 1. - (i_line)*line_axesfrac
-        baseline_next = baseline - line_axesfrac*1.
+        baseline = 1 - (i_line) * line_axesfrac
+        baseline_next = baseline - line_axesfrac * 1
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/mathtext_examples.py]: ` * 1` does nothing?

File: examples/text_labels_and_annotations/usetex_demo.py [hunk 4]
```diff
@@ -32,42 +32,46 @@
...
+         'k', linewidth=2)
 plt.text(-0.06, height - 0.06, r'$\delta$', {'color': 'k', 'fontsize': 24})
 
 # X-axis label
 plt.xticks((-1, 0, 1), ('-1', '0', '1'), color='k', size=20)
 
 # Left Y-axis labels
 plt.ylabel(r'\bf{phase field} $\phi$', {'color': 'b',
-                                          'fontsize': 20})
+                                        'fontsize': 20})
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/usetex_demo.py]: Might as well break before the `{`.

**Pull request merged**